ovh/resource_me_api_oauth2_client.go

```diff
@@ -0,0 +1,178 @@
+package ovh
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func resourceApiOauth2Client() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceApiOauth2ClientCreate,
+		ReadContext:   resourceApiOauth2ClientRead,
+		UpdateContext: resourceApiOauth2ClientUpdate,
+		DeleteContext: resourceApiOauth2ClientDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("client_id", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"callback_urls": {
+				Type:        schema.TypeList,
+				Description: "Callback URLs of the applications using this oauth2 client. Required if using the AUTHORIZATION_CODE flow.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Description: "Client ID for the oauth2 client, generated during the resource creation.",
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Description: "Secret for the oauth2 client, generated during the oauth2 client creation.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "A description of your oauth2 client.",
+				Required:    true,
+			},
+			"identity": {
+				Type:        schema.TypeString,
+				Description: "URN that will allow you to associate this oauth2 client with an access policy",
+				Computed:    true,
+			},
+			"flow": {
+				Type:         schema.TypeString,
+				ValidateFunc: helpers.ValidateEnum([]string{"AUTHORIZATION_CODE", "CLIENT_CREDENTIALS"}),
+				Description:  "OAuth2 flow type implemented for this oauth2 client. Can be either AUTHORIZATION_CODE or CLIENT_CREDENTIALS",
+				Default:      "CLIENT_CREDENTIALS",
```

Comment:
to discuss, I would find it cleaner to have this property required instead of having a default value